src/core/geometry/qgssfcgalengine.h

```diff
@@ -0,0 +1,593 @@
+   * SFCGAL context.
+   */
+  struct Deleter
+  {
+
+    /**
+     * Destroys the SFCGAL geometry \a geom, using the static QGIS
+     * SFCGAL context.
+     */
+    void CORE_EXPORT operator()( sfcgal::geometry *geom ) const;
+  };
+
+  /**
+   * Scoped SFCGAL pointer.
+   */
+  using unique_geom = std::unique_ptr< geometry, Deleter >;
+  using shared_geom = std::shared_ptr< geometry >; // NO DELETER ==> added with function make_shared_ptr!!!!!
+
+  //! Creates a shared geometry pointer with sfcgal::Deleter
+  shared_geom make_shared_geom( geometry *geom );
+
+  //! Callback uses by SFCGAL lib to push error.
+  int errorCallback( const char *, ... );
+
+  //! Callback uses by SFCGAL lib to push warning.
+  int warningCallback( const char *, ... );
+
+  /**
+   * Helper class to handle SFCGAL engine errors.
+   *
+   * Messages are hold in a stacktrace way in order to improve context understanding.
+   * \ingroup core
+   * \since QGIS 3.44
+   */
+  class CORE_EXPORT ErrorHandler
+  {
+    public:
+      //! Default constructor.
+      ErrorHandler();
+
+      /**
+      * Returns true if no failure has been caught or returns false and add new stacktrace entry.
+      *
+      * If a failure has already been caught and \a errorMsg is not null then:
+      *
+      * - a stacktrace entry is added with caller location
+      * - \a errorMsg will be updated with failure messages
+      */
+      bool hasSucceedOrStack( QString *errorMsg = nullptr, const char *fromFile = nullptr, const char *fromFunc = nullptr, int fromLine = 0 );
+
+      /**
+       * Clears failure messages and also clear \a errorMsg if not null.
+       */
+      void clearText( QString *errorMsg = nullptr );
+
+      //! Returns true if failure has been caught.
+      bool isTextEmpty() const;
+
+      //! Returns the first caught failure message.
+      QString getMainText() const;
+
+      //! Returns all failure messages as a stack trace.
+      QString getFullText() const;
+
+      //! Adds \a msg to the failure message list.
+      void addText( const QString &msg, const char *fromFile = nullptr, const char *fromFunc = nullptr, int fromLine = 0 );
+
+    private:
+      QStringList errorMessages;
+  };
+
+  //! Returns current error handler.
+  CORE_EXPORT ErrorHandler *errorHandler();
+
+  //! Shortcut for SFCGAL function definition.
+  using func_geomgeom_to_geom = sfcgal_geometry_t *( * )( const sfcgal_geometry_t *, const sfcgal_geometry_t * );
+  //! Shortcut for SFCGAL function definition.
+  using func_geom_to_geom = sfcgal_geometry_t *( * )( const sfcgal_geometry_t * );
+} // namespace sfcgal
+
+/**
+ * \ingroup core
+ * \brief Does vector analysis using the SFCGAL library and handles import, export, exception handling*
+ * \note not available in Python bindings
+ * \since QGIS 3.44
+ */
+class CORE_EXPORT QgsSfcgalEngine
+{
+  public:
+
+    /**
+     * Creates a SFGAL geometry (inherit QgsAbstractGeometry) from an internal SFCGAL geometry (from SFCGAL library).
```

Comment:
Can you please do a proof read of all of these docs and update them to match the actual behavior?

```suggestion
     * Creates a QgsAbstractGeometry from an internal SFCGAL geometry (from SFCGAL library).
```

Reply:
I meant ALL the docs. I can spot others where the content is misleading -- eg "Same as `toAbstractGeometry` but returned object is casted to QgsSfcgalGeometry."